qubes/backup.py

```diff
@@ -630,39 +630,44 @@ async def _wrap_and_send_files(self, files_to_backup, output_queue):
                 path = file_info.path
                 if callable(path):
                     path = await qubes.utils.coro_maybe(path())
-                tar_cmdline = (["tar", "-Pc", '--sparse',
-                                '-C', os.path.dirname(path)] +
-                               (['--dereference'] if
-                               file_info.subdir != "dom0-home/" else []) +
-                               ['--xform=s:^%s:%s\\0:' % (
-                                   os.path.basename(path),
-                                   file_info.subdir),
-                                   os.path.basename(path)
-                               ])
-                file_stat = os.stat(path)
-                if stat.S_ISBLK(file_stat.st_mode) or \
-                        file_info.name != os.path.basename(path):
-                    # tar doesn't handle content of block device, use our
-                    # writer
-                    # also use our tar writer when renaming file
-                    assert not stat.S_ISDIR(file_stat.st_mode), \
-                        "Renaming directories not supported"
-                    tar_cmdline = ['python3', '-m', 'qubes.tarwriter',
-                        '--override-name=%s' % (
-                            os.path.join(file_info.subdir, os.path.basename(
-                                file_info.name))),
-                        path]
-                if self.compressed:
-                    tar_cmdline.insert(-2,
-                        "--use-compress-program=%s" % self.compression_filter)
-
-                self.log.debug(" ".join(tar_cmdline))
-
-                # Pipe: tar-sparse | scrypt | tar | backup_target
-                # TODO: log handle stderr
-                # pylint: disable=not-an-iterable
-                tar_sparse = await asyncio.create_subprocess_exec(
-                    *tar_cmdline, stdout=subprocess.PIPE)
+
+                fd = os.open(path, os.O_RDONLY|o.O_NOCTTY|os.O_CLOEXEC)
```

Comment:
This won't work, `path` may point at a directory (for example for backing up dom0 home).

Reply:
`os.open()` works just fine on directories (as with C’s `open(2)`).